### ISSUE TITLE
Deploying a Function App with --deployment-container-image-name can result in a Function App that will not start.

### ISSUE THREAD

**Issue by FinVamp1** [Azure/azure-cli#21157]
> ### `az feedback` auto-generates most of the information requested below, as of CLI version 2.0.62

**Describe the bug**
The validation for --deployment-container-image-name can result in a Function App that will start as it sets the LinuxFxVersion correctly.

The linuxFxVersion should be <DockerId>/imagename:tags> 

The validation logic at https://github.com/Azure/azure-cli/blob/6c1b085a0910c6c2139006fcbd8ade44006eb6dd/src/azure-cli/azure/cli/command_modules/appservice/custom.py#L1227 doesn't seem to check this for the Functions scenarios. 

**To Reproduce**

1. Install the newest Az Cli.
2. Create a Storage Account, Resource Group and Managed Identity.
3. Run this command. 

az functionapp create --name <appNane> --resource-group <resourceGroupName> \\
 --storage-account <storageAccountName> \\
 --plan <planName \\
 --functions-version 3 \\
 --runtime python \\
 --os-type Linux \\
 --deployment-container-image-name <imageName:tags> \\
 --assign-identity <appIdentity> 

4. You will see a Function App that is created but has no Functions and returns a Server error.
5. The internal logs show the docker pull is failing/
6. Updating the Linux Fx Version to DockerId/imagename:tags works. 

**Expected behavior**
1. That the Function App starts.

**Environment summary**
Reproduced using the Azure Cloud Shell. 

**Additional context**
https://docs.microsoft.com/en-us/azure/azure-functions/functions-create-function-linux-custom-image 
https://docs.microsoft.com/en-us/azure/app-service/configure-custom-container?pivots=container-linux#change-the-docker-image-of-a-custom-container

**Comment by yonzhan** [comment]
route to CXP team to help with

**Comment by SatishBoddu-MSFT** [comment]
Hello @FinVamp1 Thanks for the context!

As we do have an internal Incident ticket created on this issue, i am routing this to the service team asap.

**Comment by pragnagopa** [comment]
We do not expect customers to set LinuxFxVersion with blessed image - combination of FUNCTIONS_EXTENSION_VERSION and `linuxFxVersion` - `runtime|version` - should work - for example - For V3 python 3.7 app 

`FUNCTIONS_EXTENSION_VERSION` should be set to `~3`
`linuxFxVersion` should be set to `python|3.7`

**Comment by zandym** [comment]
What about container-based function apps?

Running `az functionapp config set` with `--linux-fx-version 'DOCKER|<image>:<tag>'` immediately after `az functionapp create` was the only way we could successfully deploy.

**Comment by pragnagopa** [comment]
Yes, for custom containers - `az functionapp config set --linux-fx-version 'DOCKER|<image>:<tag>' is the right command.  Validation should be added in the backend to ensure docker image provided is valid - this will help all the tooling to rely on a single validation that is built into the Platform

**Comment by pragnagopa** [comment]
@FinVamp1 - let us know if you think validation should be built into App Services . We discuss more details offline.

**Comment by zandym** [comment]
@pragnagopa  - Why do we have to run the `az functionapp config set` command though? 
The `az functionapp create` with the `--deployment-container-image-name` argument has all the information it needs to set that config value behind the scenes.

Another thing... in our experience, this type of deployment using containers doesn't work on other registries besides ACR. We've tried both Gitlab and Artifactory registries using the same methodology.  It errors out with `Azure Functions runtime is unreachable` and there is nothing to go on to debug the issue.

**Comment by pragnagopa** [comment]
@zandym - Thanks for the details. I understand the issue better now.  Yes, `--deployment-container-image-name` command should set the `linuxFxVersion` and should not require user to run another command.

We will add validation if the image is valid or not on the platform.

> We've tried both Gitlab and Artifactory registries using the same methodology.

For this - please open an issue on https://github.com/Azure/azure-functions-docker with details on the AppName / appx time range. We need to take a look the logs to understand why other registries are not working.
